drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -529,6 +557,13 @@ end
 
 local function info_changed(driver, device, event, args)
   if device.profile.id ~= args.old_st_store.profile.id then
+    for cap_id, attributes in pairs(subscribed_attributes) do
```

Comment:
This should all be handled by the implementation of the `subscribe` function in the lua libs. See the `populate_subscribed_attributes_from_driver_defaults` function that is called in the `subscribe` definition in the lua libs.

Therefore, I think this can be removed and just the `subscribe()` call is needed.